data_scrapers/alameda_county.py

```diff
@@ -0,0 +1,167 @@
+    series["deaths"] = [{k: v for k, v in entry.items() if k in death_keys} for entry in re_keyed]
+
+    return series
+
+def get_notes() -> str:
+    """Scrape notes and disclaimers from dashboards."""
+    notes = ""
+    driver = webdriver.Firefox()
+    driver.implicitly_wait(30)
+    for url in dashboards:
+        driver.get(url)
+        soup = BeautifulSoup(driver.page_source,'lxml')
+        for p_tag in soup.find_all('p'):
+            if 'Notes' in p_tag.get_text():
+                notes += p_tag.get_text().strip()
+    return notes
+
+def get_demographics() -> (Dict, List):
+    """Fetch cases and deaths by age, gender, race, ethnicity
+    Returns the dictionary value for {"cases_totals": {}, "death_totals":{}}, as well as a list of 
+    strings describing datapoints that have a value of "<10". 
+    To crete a DataFrame from the dictionary, run 'pd.DataFrame(get_demographics()[0])' 
+    Note that the DataFrame will convert the "<10" strings to NaN. """
+
+    param_list = {'where': "Geography='Alameda County'", 'outFields': '*', 'outSR':4326, 'f':'json'}
+    response = requests.get(demographics, params=param_list)
+    response.raise_for_status()
+    parsed = response.json()
+    fields = parsed['fields']
+    data = parsed['features'][0]['attributes']
+
+    # copy dictionary structure of global 'out' dictionary to local variable
+    demo_totals = { "case_totals": out["case_totals"], "death_totals": out["death_totals"]} 
+
+    # Parse and re-key
+    # note: gender does not currently include other genders
+    # dict of target_label : source_label for re-keying
+    gender_keys = {"female": "Female", "male": "Male", "unknown": "Unknown_Sex"}
+    race_keys = {"Latinx_or_Hispanic": "Hispanic/Latino", "Asian": "Asian", "African_Amer": "African_American/Black",
+                 "White": "White", "Pacific_Islander": "Pacific_Islander", "Native_Amer": "Native_American", "Multiple_Race": "Multirace",
+                 "Other": "Other_Race", "Unknown": "Unknown_Race"}
+
+    # parse gender cases
+    for k, v in gender_keys.items():
+        demo_totals["case_totals"]["gender"][k] = data[v]
+    # parse race cases and deaths
+    for k, v in race_keys.items():
+        demo_totals["case_totals"]["race_eth"][k] = data[v]
+        # dashes are decoded as the escape sequence '\u2014'
+        demo_totals["death_totals"]["race_eth"][k] = data['Deaths\u2014' + v]
+    # get age cases
+    demo_totals["case_totals"]["age_group"] = {
+        k: v for k, v in data.items() if 'Age' in k}
+
+    #make a list of all datapoints with value '<10'
+    counts_lt_10 = []
+    print(demo_totals)
+    for cat, cat_dict in demo_totals.items(): # cases, deaths
+        for group, group_dict in cat_dict.items(): # dictionaries for age, race/eth
+            for key, val in group_dict.items():
+                if val == '<10':
+                    counts_lt_10.append(cat + ' ' + key)
```

Comment:
It might be helpful to include `group` here, too. Dot notation (e.g. `case_totals.race_eth.African_Amer`) might also be nicer for this use-case than just spaces.

Reply:
It looks nothing now has a value of '<10'. Sadly, maybe that is because these case and death counts have now increased past 10. They are also now using the value `null`. I've edited this logic to include group, use the dot notation, and handle `null` values.

Reply:
Oh, that’s super interesting. Is that for values they just weren’t specifying at all before? Or did `null` replace some other sort of value they used to be using?